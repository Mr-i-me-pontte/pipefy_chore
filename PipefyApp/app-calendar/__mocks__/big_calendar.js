export const label = 'Testing label';
export const onNavigate = jest.fn();
export const onViewChange = jest.fn();
export const view = 'month';
export const views = ['month', 'week', 'day', 'agenda'];

export default { label, onNavigate, onViewChange, view, views };
