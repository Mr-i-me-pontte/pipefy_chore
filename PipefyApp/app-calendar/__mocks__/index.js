import MockDate from 'mockdate';

MockDate.set('2017-12-01');

export const card = {
  due_date: '2017-12-15T16:00:00-02:00',
  id: 'aBc_1234',
  internalId: '1',
  isVisible: true,
  title: 'Testing card',
};

export const e = { preventDefault: jest.fn() };

export const event = {
  end: new Date(new Date(card.due_date).getTime() + 30 * 60000),
  id: card.id,
  start: new Date(card.due_date),
  title: card.title,
};

export const pipefy = {
  app: { pipeId: '1' },
  locale: 'en',
  openCard: jest.fn(),
  allCards: () => Promise.resolve({ cards: [card], pageInfo: { hasNextPage: false } })
};
