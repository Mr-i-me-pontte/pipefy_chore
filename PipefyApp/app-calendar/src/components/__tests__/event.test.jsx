import React from 'react';
import renderer from 'react-test-renderer';

import Event from '../event';
import { e, event, pipefy } from '../../../__mocks__';

describe('Event', () => {
  const component = renderer.create(<Event event={event} pipefy={pipefy} />);

  it('render', () => {
    expect(component.toJSON()).toMatchSnapshot();
    expect(pipefy.openCard).not.toBeCalled();
  });

  it('calls open card', () => {
    component.toJSON().props.onClick(e);
    expect(e.preventDefault).toBeCalled();
    expect(pipefy.openCard).toBeCalledWith(event.id);
  });
});
