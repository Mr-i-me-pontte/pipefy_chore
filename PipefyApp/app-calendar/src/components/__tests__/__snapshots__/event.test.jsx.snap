// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event render 1`] = `
<a
  href="#open_card"
  onClick={[Function]}
>
  Testing card
</a>
`;
